interpreter/runtime/data.ml

```diff
@@ -1,8 +1,70 @@
+open Types
+open Value
+
 type data = string ref
 type t = data
+type address = Memory.address
+
+exception Type
+exception Bounds
 
 let alloc bs = ref bs
 let size seg = I64.of_int_u (String.length !seg)
-let load seg i = (!seg).[Int64.to_int i]
+let load seg a =
+  try (!seg).[Int64.to_int a] with Invalid_argument _ -> raise Bounds
 let bytes seg = !seg
 let drop seg = seg := ""
+
+let load_bytes seg a n =
```

Comment:
Are you happy with the moderate duplication between here and `memory.ml`?

Reply:
Not really – fixed. PTAL. :)